hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPartitionedRecordLevelIndex.scala

```diff
@@ -444,6 +447,38 @@ class TestPartitionedRecordLevelIndex extends RecordLevelIndexTestBase {
     metadata.readRecordIndexLocationsWithKeys(HoodieListData.eager(recordKeys), dataTablePartition)
       .collectAsList().asScala.map(p => p.getKey -> p.getValue).toMap
   }
+
+  @ParameterizedTest
+  @EnumSource(classOf[HoodieTableType])
+  def testRLIForDeletesWithHoodieIsDeletedColumn(tableType: HoodieTableType): Unit = {
```

Comment:
Should we also add test variant(s) on the merge mode of `COMMIT_TIME_ORDERING`?

Reply:
I've added a new case `testRLIForDeletesWithCommitTimeOrdering`